MobileApplication/MobileApplication/ViewModels/CategoryViewModel.cs

```diff
@@ -77,13 +77,16 @@ private async void OnAddItem(object obj)
             await Shell.Current.GoToAsync(nameof(NewItemPage));
         }
 
-        async void OnItemSelected(Category item)
+        async void OnItemSelected(Category category)
         {
-            if (item == null)
+            if (category == null)
                 return;
 
             // This will push the ItemDetailPage onto the navigation stack
-            await Shell.Current.GoToAsync($"{nameof(ItemDetailPage)}?{nameof(ItemDetailViewModel.ItemId)}={item.Id}");
+            Item item = new Item(category.Id.ToString(),category.Title,category.DedicatedAmount.ToString());
+            var itemDetailPage = new ItemDetailPage();
+            itemDetailPage.BindingContext = item;
```

Comment:
I'm not sure it's good practice to specify `BindingContext` here because it's `CategoryViewModel`. The category should be passed to Page as a parameter. _And not the Item!_

Reply:
I used this example https://docs.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/navigation/hierarchical#passing-data-through-a-bindingcontext so I don't think it needs to be changed